### DISCUSSION TITLE
how to run lighthouse-puppeteer in Desktop mode?

### DISCUSSION THREAD

**Discussion by qkrdlsrnjs11** [GoogleChrome/lighthouse#13789] [General] [answered]
**Summary**
how to run lighthouse-puppeteer in Desktop mode?

**detailed**
I'm trying to test using this basic code of lighthouse. but If I run this code, There are shown with "Mobile" mode.(m.naver.com) (I wrote a "www.naver.com" but, m.naver.com was redirected)

I want to run at Desktop. Where do I revise the code? I try to do my best to find a solution, but I couldn't. So Please share your solution. Thanks.

my test code is same with sample code :
General Process > option1) https://github.com/GoogleChrome/lighthouse/blob/master/docs/puppeteer.md

```
const puppeteer = require('puppeteer');
const lighthouse = require('lighthouse');
const {URL} = require('url');

(async() => {
const url = 'https://chromestatus.com/features';

// Use Puppeteer to launch headful Chrome and don't use its default 800x600 viewport.
const browser = await puppeteer.launch({
  headless: false,
  defaultViewport: null,
});

// Wait for Lighthouse to open url, then inject our stylesheet.
browser.on('targetchanged', async target => {
  const page = await target.page();
  if (page && page.url() === url) {
    await page.addStyleTag({content: '* {color: red}'});
  }
});

// Lighthouse will open the URL.
// Puppeteer will observe `targetchanged` and inject our stylesheet.
const {lhr} = await lighthouse(url, {
  port: (new URL(browser.wsEndpoint())).port,
  output: 'json',
  logLevel: 'info',
});

console.log(`Lighthouse scores: ${Object.values(lhr.categories).map(c => c.score).join(', ')}`);

await browser.close();
})();
```

![스크린샷 2022-03-28 오전 9 03 15](https://user-images.githubusercontent.com/12168038/160307042-20aadedf-73c0-4305-a4d9-3d3a85f2765f.png)

**Comment by adamraine** [comment, marked as answer]
Lighthouse applies its own emulation settings to the page regardless of the environment you set up with puppeteer beforehand. To get Lighthouse to use desktop emulation, you can use the Lighthouse desktop config:

```js
const desktopConfig = require('lighthouse/lighthouse-core/config/desktop-config.js');
const {lhr} = await lighthouse(url, flags, desktopConfig);
```

**Reply by qkrdlsrnjs11** [reply]
@adamraine  I haven't find any reference related to this. Does it noted in NPM Lighthouse description?
So anyway Thank you for solution. It works well.

**Reply by adamraine** [reply]
It's mentioned in some of our docs (e.g. https://github.com/GoogleChrome/lighthouse/blob/master/docs/emulation.md)

**Reply by bharathkr1** [reply]
I just want to know the full code to Run lighthouse in web mode.

**Reply by bharathkr1** [reply]
@adamraine Plz help

**Reply by qkrdlsrnjs11** [reply]
so do I

**Reply by bharathkr1** [reply]
@qkrdlsrnjs11  could u help me on this

**Reply by adamraine** [reply]
Here is a standalone code snippet that should work. Please read our [docs](https://github.com/GoogleChrome/lighthouse/tree/master/docs) for more help.
```js
const lighthouse = require('lighthouse');
const desktopConfig = require('lighthouse/lighthouse-core/config/desktop-config.js');
const {lhr} = await lighthouse('https://example.com', undefined, desktopConfig);
```

**Reply by bharathkr1** [reply]
Thanks a lot @adamraine
